ili/validation/metrics.py

```diff
@@ -70,16 +74,79 @@ def _build_sampler(self, posterior) -> _BaseSampler:
             if self.backend != 'sbi':
                 raise ValueError(
                     'Pyro backend is only available for sbi posteriors')
+                    
             # check if DirectPosterior is available
+            # First case: we have a NeuralPosteriorEnsemble instance
             if isinstance(posterior, NeuralPosteriorEnsemble):
+                # We only need to check the first element (one model class for a posterior ensemble)
                 if isinstance(posterior.posteriors[0], DirectPosterior):
```

Comment:
We should probably remove this. Sometimes, people may want to use mcmc for SNPE methods...

Reply:
Done. Now, we only use DirectPosteriors if you specify `'sample_method': 'direct'`